### PR TITLE
III-3773 Add PHPStan

### PR DESCRIPTION
**Pull request by LucWollants**
### Added
- Add PHPStan

**Note**: not really part of ticket III-3773 but it works very well with strict types and was minimal work

**Note**: based on https://github.com/cultuurnet/search-v3/pull/49

---
Ticket: https://jira.uitdatabank.be/browse/III-3773

### CODE REVIEW

**Comment by LucWollants** [comment]
@bertramakers this requires a change to the GitHub setting to enforce passing PHPStan check.

File: src/Serializer/SerializerInterface.php [hunk 1]
```diff
@@ -23,8 +23,8 @@ public function serialize($object);
      * Deserialize the jsonld.
      * @param string $jsonString
      *   The json formatted string of the object.
-     * @param $class
+     * @param string $class
      *   Type to decode to.
      */
-    public function deserialize(string $jsonString, $class = PagedCollection::class);
+    public function deserialize(string $jsonString, string $class = PagedCollection::class);
```
Comment (jdrieghe) [inline src/Serializer/SerializerInterface.php:29]: https://phpstan.org/writing-php-code/phpdoc-types#class-string

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -46,3 +46,23 @@ jobs:
...
+      name: Code style
+      steps:
+        - name: 📤 Checkout project
+          uses: actions/checkout@v2
+
+        - name: 🐘 Install PHP
+          uses: shivammathur/setup-php@v2
+          with:
+            php-version: 7.1
+            extensions: intl, tidy, bcmath, zip
```
Comment (jdrieghe) [inline .github/workflows/ci.yml:61]: do we need all these extensions for `search-v3`?

Comment (LucWollants) [reply .github/workflows/ci.yml:61]: I can try without them, I thought they were needed for PHPStan?

Comment (bertramakers) [reply .github/workflows/ci.yml:61]: `zip` seems especially weird 🤔 
Maybe try without them and re-add those that are necessary? And maybe we should also add those that are required to `composer.json` then?

**Comment by bertramakers** [comment]
@LucWollants I think the base branch of this PR should be #49 and not `main`?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (LucWollants) [reply .github/workflows/ci.yml]: I will re-add the PHP extensions one by one.

Comment (bertramakers) [reply .github/workflows/ci.yml]: Seems like none of them are needed. Makes sense for this package I think. Should we double-check other repositories as well?

Comment (LucWollants) [reply .github/workflows/ci.yml]: I can do that yes. Maybe I take this list?

- https://github.com/cultuurnet/udb3-silex
- https://github.com/cultuurnet/calendar-summary-v3
- https://github.com/cultuurnet/search-v3
- https://github.com/cultuurnet/udb3-search-service
- https://github.com/cultuurnet/jwt-provider
- https://github.com/cultuurnet/udb3-api-guard

**Review by jdrieghe — APPROVED** [review]

**Pull request merged**